<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <title>Test Report</title>
    <link href="assets/style.css" rel="stylesheet" type="text/css"/></head>
  <body onLoad="init()">
    <script>/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */


function toArray(iter) {
    if (iter === null) {
        return null;
    }
    return Array.prototype.slice.call(iter);
}

function find(selector, elem) { // eslint-disable-line no-redeclare
    if (!elem) {
        elem = document;
    }
    return elem.querySelector(selector);
}

function findAll(selector, elem) {
    if (!elem) {
        elem = document;
    }
    return toArray(elem.querySelectorAll(selector));
}

function sortColumn(elem) {
    toggleSortStates(elem);
    const colIndex = toArray(elem.parentNode.childNodes).indexOf(elem);
    let key;
    if (elem.classList.contains('result')) {
        key = keyResult;
    } else if (elem.classList.contains('links')) {
        key = keyLink;
    } else {
        key = keyAlpha;
    }
    sortTable(elem, key(colIndex));
}

function showAllExtras() { // eslint-disable-line no-unused-vars
    findAll('.col-result').forEach(showExtras);
}

function hideAllExtras() { // eslint-disable-line no-unused-vars
    findAll('.col-result').forEach(hideExtras);
}

function showExtras(colresultElem) {
    const extras = colresultElem.parentNode.nextElementSibling;
    const expandcollapse = colresultElem.firstElementChild;
    extras.classList.remove('collapsed');
    expandcollapse.classList.remove('expander');
    expandcollapse.classList.add('collapser');
}

function hideExtras(colresultElem) {
    const extras = colresultElem.parentNode.nextElementSibling;
    const expandcollapse = colresultElem.firstElementChild;
    extras.classList.add('collapsed');
    expandcollapse.classList.remove('collapser');
    expandcollapse.classList.add('expander');
}

function showFilters() {
    const filterItems = document.getElementsByClassName('filter');
    for (let i = 0; i < filterItems.length; i++)
        filterItems[i].hidden = false;
}

function addCollapse() {
    // Add links for show/hide all
    const resulttable = find('table#results-table');
    const showhideall = document.createElement('p');
    showhideall.innerHTML = '<a href="javascript:showAllExtras()">Show all details</a> / ' +
                            '<a href="javascript:hideAllExtras()">Hide all details</a>';
    resulttable.parentElement.insertBefore(showhideall, resulttable);

    // Add show/hide link to each result
    findAll('.col-result').forEach(function(elem) {
        const collapsed = getQueryParameter('collapsed') || 'Passed';
        const extras = elem.parentNode.nextElementSibling;
        const expandcollapse = document.createElement('span');
        if (extras.classList.contains('collapsed')) {
            expandcollapse.classList.add('expander');
        } else if (collapsed.includes(elem.innerHTML)) {
            extras.classList.add('collapsed');
            expandcollapse.classList.add('expander');
        } else {
            expandcollapse.classList.add('collapser');
        }
        elem.appendChild(expandcollapse);

        elem.addEventListener('click', function(event) {
            if (event.currentTarget.parentNode.nextElementSibling.classList.contains('collapsed')) {
                showExtras(event.currentTarget);
            } else {
                hideExtras(event.currentTarget);
            }
        });
    });
}

function getQueryParameter(name) {
    const match = RegExp('[?&]' + name + '=([^&]*)').exec(window.location.search);
    return match && decodeURIComponent(match[1].replace(/\+/g, ' '));
}

function init () { // eslint-disable-line no-unused-vars
    resetSortHeaders();

    addCollapse();

    showFilters();

    sortColumn(find('.initial-sort'));

    findAll('.sortable').forEach(function(elem) {
        elem.addEventListener('click',
            function() {
                sortColumn(elem);
            }, false);
    });
}

function sortTable(clicked, keyFunc) {
    const rows = findAll('.results-table-row');
    const reversed = !clicked.classList.contains('asc');
    const sortedRows = sort(rows, keyFunc, reversed);
    /* Whole table is removed here because browsers acts much slower
     * when appending existing elements.
     */
    const thead = document.getElementById('results-table-head');
    document.getElementById('results-table').remove();
    const parent = document.createElement('table');
    parent.id = 'results-table';
    parent.appendChild(thead);
    sortedRows.forEach(function(elem) {
        parent.appendChild(elem);
    });
    document.getElementsByTagName('BODY')[0].appendChild(parent);
}

function sort(items, keyFunc, reversed) {
    const sortArray = items.map(function(item, i) {
        return [keyFunc(item), i];
    });

    sortArray.sort(function(a, b) {
        const keyA = a[0];
        const keyB = b[0];

        if (keyA == keyB) return 0;

        if (reversed) {
            return keyA < keyB ? 1 : -1;
        } else {
            return keyA > keyB ? 1 : -1;
        }
    });

    return sortArray.map(function(item) {
        const index = item[1];
        return items[index];
    });
}

function keyAlpha(colIndex) {
    return function(elem) {
        return elem.childNodes[1].childNodes[colIndex].firstChild.data.toLowerCase();
    };
}

function keyLink(colIndex) {
    return function(elem) {
        const dataCell = elem.childNodes[1].childNodes[colIndex].firstChild;
        return dataCell == null ? '' : dataCell.innerText.toLowerCase();
    };
}

function keyResult(colIndex) {
    return function(elem) {
        const strings = ['Error', 'Failed', 'Rerun', 'XFailed', 'XPassed',
            'Skipped', 'Passed'];
        return strings.indexOf(elem.childNodes[1].childNodes[colIndex].firstChild.data);
    };
}

function resetSortHeaders() {
    findAll('.sort-icon').forEach(function(elem) {
        elem.parentNode.removeChild(elem);
    });
    findAll('.sortable').forEach(function(elem) {
        const icon = document.createElement('div');
        icon.className = 'sort-icon';
        icon.textContent = 'vvv';
        elem.insertBefore(icon, elem.firstChild);
        elem.classList.remove('desc', 'active');
        elem.classList.add('asc', 'inactive');
    });
}

function toggleSortStates(elem) {
    //if active, toggle between asc and desc
    if (elem.classList.contains('active')) {
        elem.classList.toggle('asc');
        elem.classList.toggle('desc');
    }

    //if inactive, reset all other functions and add ascending active
    if (elem.classList.contains('inactive')) {
        resetSortHeaders();
        elem.classList.remove('inactive');
        elem.classList.add('active');
    }
}

function isAllRowsHidden(value) {
    return value.hidden == false;
}

function filterTable(elem) { // eslint-disable-line no-unused-vars
    const outcomeAtt = 'data-test-result';
    const outcome = elem.getAttribute(outcomeAtt);
    const classOutcome = outcome + ' results-table-row';
    const outcomeRows = document.getElementsByClassName(classOutcome);

    for(let i = 0; i < outcomeRows.length; i++){
        outcomeRows[i].hidden = !elem.checked;
    }

    const rows = findAll('.results-table-row').filter(isAllRowsHidden);
    const allRowsHidden = rows.length == 0 ? true : false;
    const notFoundMessage = document.getElementById('not-found-message');
    notFoundMessage.hidden = !allRowsHidden;
}
</script>
    <h1>report.html</h1>
    <p>Report generated on 22-Aug-2022 at 06:28:37 by <a href="https://pypi.python.org/pypi/pytest-html">pytest-html</a> v3.1.1</p>
    <h2>Environment</h2>
    <table id="environment">
      <tr>
        <td>CI</td>
        <td>true</td></tr>
      <tr>
        <td>JAVA_HOME</td>
        <td>/usr/lib/jvm/temurin-11-jdk-amd64</td></tr>
      <tr>
        <td>Packages</td>
        <td>{"pluggy": "1.0.0", "py": "1.11.0", "pytest": "6.2.5"}</td></tr>
      <tr>
        <td>Platform</td>
        <td>Linux-5.15.0-1017-azure-x86_64-with-glibc2.2.5</td></tr>
      <tr>
        <td>Plugins</td>
        <td>{"cov": "2.12.1", "html": "3.1.1", "metadata": "2.0.2", "sugar": "0.9.4"}</td></tr>
      <tr>
        <td>Python</td>
        <td>3.8.13</td></tr></table>
    <h2>Summary</h2>
    <p>65 tests ran in 524.17 seconds. </p>
    <p class="filter" hidden="true">(Un)check the boxes to filter the results.</p><input checked="true" class="filter" data-test-result="passed" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="passed">65 passed</span>, <input checked="true" class="filter" data-test-result="skipped" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="skipped">1 skipped</span>, <input checked="true" class="filter" data-test-result="failed" disabled="true" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="failed">0 failed</span>, <input checked="true" class="filter" data-test-result="error" disabled="true" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="error">0 errors</span>, <input checked="true" class="filter" data-test-result="xfailed" disabled="true" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="xfailed">0 expected failures</span>, <input checked="true" class="filter" data-test-result="xpassed" disabled="true" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="xpassed">0 unexpected passes</span>
    <h2>Results</h2>
    <table id="results-table">
      <thead id="results-table-head">
        <tr>
          <th class="sortable result initial-sort" col="result">Result</th>
          <th class="sortable" col="name">Test</th>
          <th class="sortable" col="duration">Duration</th>
          <th class="sortable links" col="links">Links</th></tr>
        <tr hidden="true" id="not-found-message">
          <th colspan="4">No results found. Try to check the filters</th></tr></thead>
      <tbody class="skipped results-table-row">
        <tr>
          <td class="col-result">Skipped</td>
          <td class="col-name">tests/processor/test_test_command_callback.py::test_check_coverage</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">(&#x27;/home/runner/work/defects4cpp/defects4cpp/tests/processor/test_test_command_callback.py&#x27;, 199, &#x27;Skipped: Temporarily disabled. Should be ran on Docker.&#x27;)<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/message/test_message.py::test_change_logger_path</td>
          <td class="col-duration">0.01</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>[test_change_logger_path] cloning a new repository from &#x27;test-url&#x27;
[test_change_logger_path] git-clone failed.
                Command: git clone -v -c core.autocrlf=false test-url /home/runner/work/defects4cpp/defects4cpp/mydir/test_change_logger_path0/test_change_logger_path/test_change_logger_path/.repo
                Status: 128
                Path: (nonexistent) /home/runner/work/defects4cpp/defects4cpp/mydir/test_change_logger_path0/test_change_logger_path/test_change_logger_path/.repo
                Metadata: test_change_logger_path

[test_change_logger_path] cloning a new repository from &#x27;test-url&#x27;
[test_change_logger_path] git-clone failed.
                Command: git clone -v -c core.autocrlf=false test-url /home/runner/work/defects4cpp/defects4cpp/mydir/test_change_logger_path0/test_change_logger_path/test_change_logger_path/.repo
                Status: 128
                Path: (nonexistent) /home/runner/work/defects4cpp/defects4cpp/mydir/test_change_logger_path0/test_change_logger_path/test_change_logger_path/.repo
                Metadata: test_change_logger_path

<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/message/test_message.py::test_output_log_to_stdout</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_argparse.py::test_read_config_not_exist</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_argparse.py::test_read_config_invalid_json</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_argparse.py::test_read_config_corrupted_json</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_argparse.py::test_read_config</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_argparse.py::test_write_config</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_argparse.py::test_project_parser_invalid_project_should_throw</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_argparse.py::test_project_parser_should_read_config_json</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_argparse.py::test_vcs_parser_invalid_project_should_throw</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stderr call------------------------------ <br/>usage: __main__.py [-h] [--compilation_db_tool COMPILATION_DB_TOOL] [-b]
                   [-t WORKSPACE]
                   {coreutils,cpp_peglib,cppcheck,dlt_daemon,example,exiv2,jerryscript,libchewing,libssh,libtiff,libtiff_sanitizer,libucl,libxml2,md4c,ndpi,openssl,proj,wget2,wireshark,xbps,yaml_cpp,yara,zsh}
                   index
__main__.py: error: argument project: invalid choice: &#x27;foobar&#x27; (choose from &#x27;coreutils&#x27;, &#x27;cpp_peglib&#x27;, &#x27;cppcheck&#x27;, &#x27;dlt_daemon&#x27;, &#x27;example&#x27;, &#x27;exiv2&#x27;, &#x27;jerryscript&#x27;, &#x27;libchewing&#x27;, &#x27;libssh&#x27;, &#x27;libtiff&#x27;, &#x27;libtiff_sanitizer&#x27;, &#x27;libucl&#x27;, &#x27;libxml2&#x27;, &#x27;md4c&#x27;, &#x27;ndpi&#x27;, &#x27;openssl&#x27;, &#x27;proj&#x27;, &#x27;wget2&#x27;, &#x27;wireshark&#x27;, &#x27;xbps&#x27;, &#x27;yaml_cpp&#x27;, &#x27;yara&#x27;, &#x27;zsh&#x27;)
<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_argparse.py::test_vcs_parser_unordered_arguments_should_be_handled[yara 1 --buggy]</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_argparse.py::test_vcs_parser_unordered_arguments_should_be_handled[yara --buggy 1]</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_argparse.py::test_vcs_parser_unordered_arguments_should_be_handled_with_target_option</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_argparse.py::test_common_project_parser_env_option</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_argparse.py::test_common_command_parser_settings_option</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_build_command.py::test_build_command_should_generate_command_based_on_options[True-True-coverage_command bear]</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_build_command.py::test_build_command_should_generate_command_based_on_options[True-False-coverage_command]</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_build_command.py::test_build_command_should_generate_command_based_on_options[False-True-bear command]</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_build_command.py::test_build_command_should_generate_command_based_on_options[False-False-command]</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_build_command.py::test_build_command_export_commands[yara]</td>
          <td class="col-duration">72.54</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>[yara] cloning a new repository from &#x27;https://github.com/VirusTotal/yara&#x27;
[yara] checking out &#x27;cb787cc0c7edd91d245175af439a9190da97228b&#x27;
[yara] done
[yara] start building
  Starting container
    Image: hschoe/defects4cpp-ubuntu:yara
Step 1/1 : FROM hschoe/defects4cpp-ubuntu:yara

 ---&gt; fadde7191429
Successfully built fadde7191429
Successfully tagged hschoe/defects4cpp-ubuntu:yara
  Closing container
[yara] done
<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     processor.checkout:(unknown file):0 git-clone &#x27;yara&#x27;
INFO     processor.checkout:(unknown file):0 cloning yara into /home/runner/work/defects4cpp/defects4cpp/mydir/test_build_command_export_comm0/yara/.repo
INFO     processor.checkout:(unknown file):0 git-checkout
INFO     processor.checkout:(unknown file):0 git-am
INFO     processor.checkout:(unknown file):0 0001-split.patch
INFO     processor.checkout:(unknown file):0 creating &#x27;.defects4cpp.json&#x27; at /home/runner/work/defects4cpp/defects4cpp/mydir/test_build_command_export_comm0/yara/fixed#1
INFO     processor.build:(unknown file):0 &#x27;yara&#x27;
INFO     processor.core.docker:(unknown file):0 container.__enter__ (yara-dpp)
                - from drwxrwxrwx /home/runner/work/defects4cpp/defects4cpp/mydir/test_build_command_export_comm0/yara/fixed#1
                - to /home/workspace
                - mode rw
INFO     processor.core.docker:(unknown file):0 container.__exit__ (yara-dpp)
INFO     processor.build:(unknown file):0 export to &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_build_command_export_comm0&#x27;
INFO     processor.build:(unknown file):0 done<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_build_command.py::test_build_command_export_commands[cppcheck]</td>
          <td class="col-duration">336.96</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>[cppcheck] cloning a new repository from &#x27;https://github.com/danmar/cppcheck.git&#x27;
[cppcheck] checking out &#x27;007b5d3e8d6d83237d425fdddd0c40089dcab59b&#x27;
[cppcheck] done
[cppcheck] start building
  Starting container
    Image: hschoe/defects4cpp-ubuntu:cppcheck
Step 1/1 : FROM hschoe/defects4cpp-ubuntu:cppcheck

 ---&gt; 80942fa47610
Successfully built 80942fa47610
Successfully tagged hschoe/defects4cpp-ubuntu:cppcheck
  Closing container
[cppcheck] done
<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     processor.checkout:(unknown file):0 git-clone &#x27;cppcheck&#x27;
INFO     processor.checkout:(unknown file):0 cloning cppcheck into /home/runner/work/defects4cpp/defects4cpp/mydir/test_build_command_export_comm1/cppcheck/.repo
INFO     processor.checkout:(unknown file):0 git-checkout
INFO     processor.checkout:(unknown file):0 git-am
INFO     processor.checkout:(unknown file):0 no patches
INFO     processor.checkout:(unknown file):0 creating &#x27;.defects4cpp.json&#x27; at /home/runner/work/defects4cpp/defects4cpp/mydir/test_build_command_export_comm1/cppcheck/fixed#1
INFO     processor.build:(unknown file):0 &#x27;cppcheck&#x27;
INFO     processor.core.docker:(unknown file):0 container.__enter__ (cppcheck-dpp)
                - from drwxrwxrwx /home/runner/work/defects4cpp/defects4cpp/mydir/test_build_command_export_comm1/cppcheck/fixed#1
                - to /home/workspace
                - mode rw
INFO     processor.core.docker:(unknown file):0 container.__exit__ (cppcheck-dpp)
INFO     processor.build:(unknown file):0 export to &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_build_command_export_comm1&#x27;
INFO     processor.build:(unknown file):0 done<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_build_command.py::test_build_command_rebuild_image</td>
          <td class="col-duration">26.06</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> -------------------------------Captured log call-------------------------------- <br/>INFO     processor.build:(unknown file):0 &#x27;test_build_command_rebuild_image&#x27;
INFO     processor.core.docker:(unknown file):0 container.__enter__ (test_build_command_rebuild_image-dpp)
                - from drwxr-xr-x /home/runner/work/defects4cpp/defects4cpp/mydir/test_build_command_rebuild_ima0/test_build_command_rebuild_image/test_build_command_rebuild_image/fixed#1
                - to /home/workspace
                - mode rw
INFO     processor.core.docker:(unknown file):0 container.__exit__ (test_build_command_rebuild_image-dpp)
INFO     processor.build:(unknown file):0 done
INFO     processor.build:(unknown file):0 &#x27;test_build_command_rebuild_image&#x27;
INFO     processor.core.docker:(unknown file):0 container.__enter__ (test_build_command_rebuild_image-dpp)
                - from drwxr-xr-x /home/runner/work/defects4cpp/defects4cpp/mydir/test_build_command_rebuild_ima0/test_build_command_rebuild_image/test_build_command_rebuild_image/fixed#1
                - to /home/workspace
                - mode rw
INFO     processor.core.docker:(unknown file):0 container.__exit__ (test_build_command_rebuild_image-dpp)
INFO     processor.build:(unknown file):0 done<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_case_expression.py::test_validate_case</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_case_expression.py::test_invalid_case_expression</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_case_expression.py::test_no_case_is_provided</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_case_expression.py::test_exclude_only</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_checkout.py::test_git_clone_error</td>
          <td class="col-duration">0.01</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>[test_git_clone_error] cloning a new repository from &#x27;test-url&#x27;
[test_git_clone_error] git-clone failed.
                Command: git clone -v -c core.autocrlf=false test-url /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_clone_error0/test_git_clone_error/test_git_clone_error/.repo
                Status: 128
                Path: (nonexistent) /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_clone_error0/test_git_clone_error/test_git_clone_error/.repo
                Metadata: test_git_clone_error

<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     processor.checkout:(unknown file):0 git-clone &#x27;test_git_clone_error&#x27;
INFO     processor.checkout:(unknown file):0 cloning test_git_clone_error into /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_clone_error0/test_git_clone_error/test_git_clone_error/.repo
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;clone&#x27;, &#x27;-v&#x27;, &#x27;-c&#x27;, &#x27;core.autocrlf=false&#x27;, &#x27;test-url&#x27;, &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_clone_error0/test_git_clone_error/test_git_clone_error/.repo&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp, universal_newlines=True, shell=None, istream=None)
DEBUG    git.repo.base:(unknown file):0 Cmd([&#x27;git&#x27;, &#x27;clone&#x27;, &#x27;-v&#x27;, &#x27;-c&#x27;, &#x27;core.autocrlf=false&#x27;, &#x27;test-url&#x27;, &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_clone_error0/test_git_clone_error/test_git_clone_error/.repo&#x27;])&#x27;s unused stdout: 
DEBUG    git.cmd:(unknown file):0 AutoInterrupt wait stderr: b&quot;fatal: repository &#x27;test-url&#x27; does not exist\n&quot;
ERROR    processor.checkout:(unknown file):0 git-clone failed.
                Command: git clone -v -c core.autocrlf=false test-url /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_clone_error0/test_git_clone_error/test_git_clone_error/.repo
                Status: 128
                Path: (nonexistent) /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_clone_error0/test_git_clone_error/test_git_clone_error/.repo
                Metadata: test_git_clone_error<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_checkout.py::test_git_worktree_error</td>
          <td class="col-duration">0.42</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>[test_git_worktree_error] cloning a new repository from &#x27;https://github.com/Suresoft-GLaDOS/defects4cpp-test&#x27;
[test_git_worktree_error] checking out &#x27;1&#x27;
[test_git_worktree_error] git-worktree failed.
                Path: (nonexistent) /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_worktree_error0/test_git_worktree_error/test_git_worktree_error/fixed#1
                Defect: Defect(hash=&#x27;1&#x27;, buggy_patch=&#x27;&#x27;, fixed_patch=&#x27;&#x27;, common_patch=&#x27;&#x27;, split_patch=&#x27;&#x27;, num_cases=1, case=[1], description=&#x27;test&#x27;, extra_tests=[])

<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     processor.checkout:(unknown file):0 git-clone &#x27;test_git_worktree_error&#x27;
INFO     processor.checkout:(unknown file):0 cloning test_git_worktree_error into /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_worktree_error0/test_git_worktree_error/test_git_worktree_error/.repo
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;clone&#x27;, &#x27;-v&#x27;, &#x27;-c&#x27;, &#x27;core.autocrlf=false&#x27;, &#x27;https://github.com/Suresoft-GLaDOS/defects4cpp-test&#x27;, &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_worktree_error0/test_git_worktree_error/test_git_worktree_error/.repo&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp, universal_newlines=True, shell=None, istream=None)
DEBUG    git.repo.base:(unknown file):0 Cmd([&#x27;git&#x27;, &#x27;clone&#x27;, &#x27;-v&#x27;, &#x27;-c&#x27;, &#x27;core.autocrlf=false&#x27;, &#x27;https://github.com/Suresoft-GLaDOS/defects4cpp-test&#x27;, &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_worktree_error0/test_git_worktree_error/test_git_worktree_error/.repo&#x27;])&#x27;s unused stdout: 
INFO     processor.checkout:(unknown file):0 git-checkout
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;worktree&#x27;, &#x27;add&#x27;, &#x27;-f&#x27;, &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_worktree_error0/test_git_worktree_error/test_git_worktree_error/fixed#1&#x27;, &#x27;1&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_worktree_error0/test_git_worktree_error/test_git_worktree_error/.repo, universal_newlines=False, shell=None, istream=None)
ERROR    processor.checkout:(unknown file):0 git-worktree failed.
                Path: (nonexistent) /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_worktree_error0/test_git_worktree_error/test_git_worktree_error/fixed#1
                Defect: Defect(hash=&#x27;1&#x27;, buggy_patch=&#x27;&#x27;, fixed_patch=&#x27;&#x27;, common_patch=&#x27;&#x27;, split_patch=&#x27;&#x27;, num_cases=1, case=[1], description=&#x27;test&#x27;, extra_tests=[])<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_checkout.py::test_git_checkout_error</td>
          <td class="col-duration">0.48</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>[test_git_checkout_error] cloning a new repository from &#x27;https://github.com/Suresoft-GLaDOS/defects4cpp-test&#x27;
[test_git_checkout_error] checking out &#x27;0a158cb95d7ed8e64552ef80df6e6204205d4fa5&#x27;
[test_git_checkout_error] git-checkout failed (not a git repository).
                Path: drwxr-xr-x /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_checkout_error0/test_git_checkout_error/test_git_checkout_error/fixed#1
                Defect: Defect(hash=&#x27;0a158cb95d7ed8e64552ef80df6e6204205d4fa5&#x27;, buggy_patch=&#x27;&#x27;, fixed_patch=&#x27;&#x27;, common_patch=&#x27;&#x27;, split_patch=&#x27;&#x27;, num_cases=1, case=[1], description=&#x27;test&#x27;, extra_tests=[])

<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     processor.checkout:(unknown file):0 git-clone &#x27;test_git_checkout_error&#x27;
INFO     processor.checkout:(unknown file):0 cloning test_git_checkout_error into /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_checkout_error0/test_git_checkout_error/test_git_checkout_error/.repo
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;clone&#x27;, &#x27;-v&#x27;, &#x27;-c&#x27;, &#x27;core.autocrlf=false&#x27;, &#x27;https://github.com/Suresoft-GLaDOS/defects4cpp-test&#x27;, &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_checkout_error0/test_git_checkout_error/test_git_checkout_error/.repo&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp, universal_newlines=True, shell=None, istream=None)
DEBUG    git.repo.base:(unknown file):0 Cmd([&#x27;git&#x27;, &#x27;clone&#x27;, &#x27;-v&#x27;, &#x27;-c&#x27;, &#x27;core.autocrlf=false&#x27;, &#x27;https://github.com/Suresoft-GLaDOS/defects4cpp-test&#x27;, &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_checkout_error0/test_git_checkout_error/test_git_checkout_error/.repo&#x27;])&#x27;s unused stdout: 
INFO     processor.checkout:(unknown file):0 git-checkout
ERROR    processor.checkout:(unknown file):0 git-checkout failed (not a git repository).
                Path: drwxr-xr-x /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_checkout_error0/test_git_checkout_error/test_git_checkout_error/fixed#1
                Defect: Defect(hash=&#x27;0a158cb95d7ed8e64552ef80df6e6204205d4fa5&#x27;, buggy_patch=&#x27;&#x27;, fixed_patch=&#x27;&#x27;, common_patch=&#x27;&#x27;, split_patch=&#x27;&#x27;, num_cases=1, case=[1], description=&#x27;test&#x27;, extra_tests=[])<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_checkout.py::test_git_submodule_init_error</td>
          <td class="col-duration">2.22</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>[test_git_submodule_init_error] cloning a new repository from &#x27;https://github.com/Suresoft-GLaDOS/defects4cpp-test&#x27;
[test_git_submodule_init_error] checking out &#x27;0a158cb95d7ed8e64552ef80df6e6204205d4fa5&#x27;
[test_git_submodule_init_error] git-submodule failed.
                Command: git submodule update --init
                Status: 128
                Output: 
  stderr: &#x27;Submodule &#x27;defects4cpp&#x27; (https://github.com/Suresoft-GLaDOS/defects4cpp) registered for path &#x27;defects4cpp&#x27;
Cloning into &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_submodule_init_error0/test_git_submodule_init_error/test_git_submodule_init_error/fixed#1/defects4cpp&#x27;...
fatal: remote error: upload-pack: not our ref d25dfb719fb961b7c927ddb3be886c059cd71621
fatal: Fetched in submodule path &#x27;defects4cpp&#x27;, but it did not contain d25dfb719fb961b7c927ddb3be886c059cd71621. Direct fetching of that commit failed.&#x27;

<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     processor.checkout:(unknown file):0 git-clone &#x27;test_git_submodule_init_error&#x27;
INFO     processor.checkout:(unknown file):0 cloning test_git_submodule_init_error into /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_submodule_init_error0/test_git_submodule_init_error/test_git_submodule_init_error/.repo
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;clone&#x27;, &#x27;-v&#x27;, &#x27;-c&#x27;, &#x27;core.autocrlf=false&#x27;, &#x27;https://github.com/Suresoft-GLaDOS/defects4cpp-test&#x27;, &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_submodule_init_error0/test_git_submodule_init_error/test_git_submodule_init_error/.repo&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp, universal_newlines=True, shell=None, istream=None)
DEBUG    git.repo.base:(unknown file):0 Cmd([&#x27;git&#x27;, &#x27;clone&#x27;, &#x27;-v&#x27;, &#x27;-c&#x27;, &#x27;core.autocrlf=false&#x27;, &#x27;https://github.com/Suresoft-GLaDOS/defects4cpp-test&#x27;, &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_submodule_init_error0/test_git_submodule_init_error/test_git_submodule_init_error/.repo&#x27;])&#x27;s unused stdout: 
INFO     processor.checkout:(unknown file):0 git-checkout
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;worktree&#x27;, &#x27;add&#x27;, &#x27;-f&#x27;, &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_submodule_init_error0/test_git_submodule_init_error/test_git_submodule_init_error/fixed#1&#x27;, &#x27;0a158cb95d7ed8e64552ef80df6e6204205d4fa5&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_submodule_init_error0/test_git_submodule_init_error/test_git_submodule_init_error/.repo, universal_newlines=False, shell=None, istream=None)
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;worktree&#x27;, &#x27;list&#x27;, &#x27;--porcelain&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_submodule_init_error0/test_git_submodule_init_error/test_git_submodule_init_error/.repo, universal_newlines=False, shell=None, istream=None)
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;rev-parse&#x27;, &#x27;--verify&#x27;, &#x27;HEAD&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_submodule_init_error0/test_git_submodule_init_error/test_git_submodule_init_error/fixed#1, universal_newlines=False, shell=None, istream=None)
INFO     processor.checkout:(unknown file):0 git-am
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;cat-file&#x27;, &#x27;--batch-check&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_submodule_init_error0/test_git_submodule_init_error/test_git_submodule_init_error/fixed#1, universal_newlines=False, shell=None, istream=&lt;valid stream&gt;)
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;cat-file&#x27;, &#x27;--batch&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_submodule_init_error0/test_git_submodule_init_error/test_git_submodule_init_error/fixed#1, universal_newlines=False, shell=None, istream=&lt;valid stream&gt;)
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;submodule&#x27;, &#x27;update&#x27;, &#x27;--init&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_submodule_init_error0/test_git_submodule_init_error/test_git_submodule_init_error/fixed#1, universal_newlines=False, shell=None, istream=None)
ERROR    processor.checkout:(unknown file):0 git-submodule failed.
                Command: git submodule update --init
                Status: 128
                Output: 
  stderr: &#x27;Submodule &#x27;defects4cpp&#x27; (https://github.com/Suresoft-GLaDOS/defects4cpp) registered for path &#x27;defects4cpp&#x27;
Cloning into &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_submodule_init_error0/test_git_submodule_init_error/test_git_submodule_init_error/fixed#1/defects4cpp&#x27;...
fatal: remote error: upload-pack: not our ref d25dfb719fb961b7c927ddb3be886c059cd71621
fatal: Fetched in submodule path &#x27;defects4cpp&#x27;, but it did not contain d25dfb719fb961b7c927ddb3be886c059cd71621. Direct fetching of that commit failed.&#x27;<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_checkout.py::test_git_apply_patch_error_patch_could_not_be_applied</td>
          <td class="col-duration">0.45</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>[test_git_apply_patch_error_patch_could_not_be_applied] cloning a new repository from &#x27;https://github.com/Suresoft-GLaDOS/defects4cpp-test&#x27;
[test_git_apply_patch_error_patch_could_not_be_applied] checking out &#x27;0935324da04784d625c5faef0705adf8705fab9c&#x27;
[test_git_apply_patch_error_patch_could_not_be_applied] git-am failed.
                Patch: -rw-r--r-- /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_apply_patch_error_pat0/test_git_apply_patch_error_patch_could_not_be_applied/patch/0001-common.patch
                Command: git am /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_apply_patch_error_pat0/test_git_apply_patch_error_patch_could_not_be_applied/patch/0001-common.patch
                Status: 128
                Output: 
  stderr: &#x27;error: patch failed: foo.txt:1
error: foo.txt: patch does not apply
hint: Use &#x27;git am --show-current-patch=diff&#x27; to see the failed patch&#x27;

<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     processor.checkout:(unknown file):0 git-clone &#x27;test_git_apply_patch_error_patch_could_not_be_applied&#x27;
INFO     processor.checkout:(unknown file):0 cloning test_git_apply_patch_error_patch_could_not_be_applied into /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_apply_patch_error_pat0/test_git_apply_patch_error_patch_could_not_be_applied/test_git_apply_patch_error_patch_could_not_be_applied/.repo
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;clone&#x27;, &#x27;-v&#x27;, &#x27;-c&#x27;, &#x27;core.autocrlf=false&#x27;, &#x27;https://github.com/Suresoft-GLaDOS/defects4cpp-test&#x27;, &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_apply_patch_error_pat0/test_git_apply_patch_error_patch_could_not_be_applied/test_git_apply_patch_error_patch_could_not_be_applied/.repo&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp, universal_newlines=True, shell=None, istream=None)
DEBUG    git.repo.base:(unknown file):0 Cmd([&#x27;git&#x27;, &#x27;clone&#x27;, &#x27;-v&#x27;, &#x27;-c&#x27;, &#x27;core.autocrlf=false&#x27;, &#x27;https://github.com/Suresoft-GLaDOS/defects4cpp-test&#x27;, &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_apply_patch_error_pat0/test_git_apply_patch_error_patch_could_not_be_applied/test_git_apply_patch_error_patch_could_not_be_applied/.repo&#x27;])&#x27;s unused stdout: 
INFO     processor.checkout:(unknown file):0 git-checkout
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;worktree&#x27;, &#x27;add&#x27;, &#x27;-f&#x27;, &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_apply_patch_error_pat0/test_git_apply_patch_error_patch_could_not_be_applied/test_git_apply_patch_error_patch_could_not_be_applied/fixed#1&#x27;, &#x27;0935324da04784d625c5faef0705adf8705fab9c&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_apply_patch_error_pat0/test_git_apply_patch_error_patch_could_not_be_applied/test_git_apply_patch_error_patch_could_not_be_applied/.repo, universal_newlines=False, shell=None, istream=None)
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;worktree&#x27;, &#x27;list&#x27;, &#x27;--porcelain&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_apply_patch_error_pat0/test_git_apply_patch_error_patch_could_not_be_applied/test_git_apply_patch_error_patch_could_not_be_applied/.repo, universal_newlines=False, shell=None, istream=None)
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;rev-parse&#x27;, &#x27;--verify&#x27;, &#x27;HEAD&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_apply_patch_error_pat0/test_git_apply_patch_error_patch_could_not_be_applied/test_git_apply_patch_error_patch_could_not_be_applied/fixed#1, universal_newlines=False, shell=None, istream=None)
INFO     processor.checkout:(unknown file):0 git-am
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;cat-file&#x27;, &#x27;--batch-check&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_apply_patch_error_pat0/test_git_apply_patch_error_patch_could_not_be_applied/test_git_apply_patch_error_patch_could_not_be_applied/fixed#1, universal_newlines=False, shell=None, istream=&lt;valid stream&gt;)
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;cat-file&#x27;, &#x27;--batch&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_apply_patch_error_pat0/test_git_apply_patch_error_patch_could_not_be_applied/test_git_apply_patch_error_patch_could_not_be_applied/fixed#1, universal_newlines=False, shell=None, istream=&lt;valid stream&gt;)
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;rev-parse&#x27;, &#x27;--verify&#x27;, &#x27;HEAD&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_apply_patch_error_pat0/test_git_apply_patch_error_patch_could_not_be_applied/test_git_apply_patch_error_patch_could_not_be_applied/fixed#1, universal_newlines=False, shell=None, istream=None)
INFO     processor.checkout:(unknown file):0 0001-common.patch
DEBUG    git.cmd:(unknown file):0 Popen([&#x27;git&#x27;, &#x27;am&#x27;, &#x27;/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_apply_patch_error_pat0/test_git_apply_patch_error_patch_could_not_be_applied/patch/0001-common.patch&#x27;], cwd=/home/runner/work/defects4cpp/defects4cpp/mydir/test_git_apply_patch_error_pat0/test_git_apply_patch_error_patch_could_not_be_applied/test_git_apply_patch_error_patch_could_not_be_applied/fixed#1, universal_newlines=False, shell=None, istream=None)
ERROR    processor.checkout:(unknown file):0 git-am failed.
                Patch: -rw-r--r-- /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_apply_patch_error_pat0/test_git_apply_patch_error_patch_could_not_be_applied/patch/0001-common.patch
                Command: git am /home/runner/work/defects4cpp/defects4cpp/mydir/test_git_apply_patch_error_pat0/test_git_apply_patch_error_patch_could_not_be_applied/patch/0001-common.patch
                Status: 128
                Output: 
  stderr: &#x27;error: patch failed: foo.txt:1
error: foo.txt: patch does not apply
hint: Use &#x27;git am --show-current-patch=diff&#x27; to see the failed patch&#x27;<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_checkout.py::test_checkout_command</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_docker.py::test_docker_image</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>    Image: hschoe/defects4cpp-ubuntu:yara
<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_docker.py::test_docker_container</td>
          <td class="col-duration">10.59</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>  Starting container
    Image: hschoe/defects4cpp-ubuntu:yara
  Closing container
<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     processor.core.docker:(unknown file):0 container.__enter__ (yara-dpp)
                - from (nonexistent) /home/runner/work/defects4cpp/defects4cpp/mydir/test_docker_container0/yara/fixed#1
                - to /home/workspace
                - mode rw
INFO     processor.core.docker:(unknown file):0 container.__exit__ (yara-dpp)<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_docker.py::test_docker_mount_directory</td>
          <td class="col-duration">21.14</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>  Starting container
    Image: hschoe/defects4cpp-ubuntu:yara
  Closing container
  Starting container
    Image: hschoe/defects4cpp-ubuntu:yara
  Closing container
<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     processor.core.docker:(unknown file):0 container.__enter__ (yara-dpp)
                - from drwxrwxrwx /home/runner/work/defects4cpp/defects4cpp/mydir/test_docker_mount_directory0/yara/fixed#1
                - to /home/workspace
                - mode rw
INFO     processor.core.docker:(unknown file):0 container.__exit__ (yara-dpp)
INFO     processor.core.docker:(unknown file):0 container.__enter__ (yara-dpp)
                - from drwxrwxrwx /home/runner/work/defects4cpp/defects4cpp/mydir/test_docker_mount_directory0/yara/fixed#1
                - to /home/workspace
                - mode rw
INFO     processor.core.docker:(unknown file):0 container.__exit__ (yara-dpp)<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_help.py::test_check_help_attr</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_help.py::test_check_help_help</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_help.py::test_check_help_run</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>Defects4C++: Defect Taxonomies for Automated Debugging Tools
MIT Licensed, Suresoft Technologies Inc.

* d++.py build PATH [-j|--jobs=JOBS] [--coverage] [-v|--verbose]
  [-u|--uid[=UID_DPP_DOCKER_USER] [-e|--export[=EXPORT_PATH]]
  Build project inside docker.
* d++.py checkout PROJECT INDEX [-b|--buggy] [-t|--target]
  Checkout defect taxonomy.
* d++.py help
  Display help messages.
* d++.py show [PROJECT]
  Display defect taxonomy in detail.
* d++.py test PATH [-j|--jobs=JOBS] [--coverage [--additional-gcov-
  options=ADDITIONAL_GCOV_OPTIONS]] [-v|--verbose] [-c|--case=expr]
  [--output-dir=directory]
  Run testsuite inside docker. The project must have been built
  previously.
<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_show.py::test_check_show_attr</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_show.py::test_check_show_help</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_show.py::test_check_show_run</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_taxonomy.py::test_metadata_setting_compilation_db_tool_should_remove_cmake_export_macro</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_taxonomy.py::test_metadata_build_pre_steps_with_missing_key_should_throw[DPP_BUILD_PRE_STEPS]</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_taxonomy.py::test_metadata_build_pre_steps_with_missing_key_should_throw[DPP_BUILD_POST_STEPS]</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_taxonomy.py::test_metadata_build_steps_with_invalid_value_should_throw[DPP_BUILD_PRE_STEPS]</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_taxonomy.py::test_metadata_build_steps_with_invalid_value_should_throw[DPP_BUILD_POST_STEPS]</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_taxonomy.py::test_metadata_docker_build_steps_should_be_inserted[docker]</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_taxonomy.py::test_metadata_docker_build_steps_should_be_inserted[script]</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_taxonomy.py::test_metadata_variables_should_be_replaced</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_taxonomy.py::test_extra_tests</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_test.py::test_script_command</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>  case #0
<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_test.py::test_setup_script_command</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_test.py::test_coverage_script_command</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_test.py::test_tear_down_script_command</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_test.py::test_gcov_script_command</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_test.py::test_run_gcovr_script_command</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_test.py::test_create_script_gernerator_command</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_test.py::test_help_command</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_test_command_callback.py::test_check_result</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_test_command_callback.py::test_run_command</td>
          <td class="col-duration">25.89</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>  Starting container
    Image: hschoe/defects4cpp-ubuntu:test_run_command
Step 1/5 : FROM ubuntu:20.04

 ---&gt; 3bc6e9f30f51
Step 2/5 : RUN useradd --uid 1001 --home-dir /home/workspace --shell /bin/bash defects4cpp

 ---&gt; Running in 815e4f2e13eb
Removing intermediate container 815e4f2e13eb
 ---&gt; e33861a87ef0
Step 3/5 : USER defects4cpp

 ---&gt; Running in 78525aff8467
Removing intermediate container 78525aff8467
 ---&gt; aac010ec825b
Step 4/5 : ENV USER defects4cpp

 ---&gt; Running in 0af18a693558
Removing intermediate container 0af18a693558
 ---&gt; c1afafc3189e
Step 5/5 : WORKDIR /home/workspace

 ---&gt; Running in 844f49f6d413
Removing intermediate container 844f49f6d413
 ---&gt; 5794be79f75a
Successfully built 5794be79f75a
Successfully tagged hschoe/defects4cpp-ubuntu:test_run_command
  Closing container
  Starting container
    Image: hschoe/defects4cpp-ubuntu:test_run_command
  Closing container
<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     processor.core.docker:(unknown file):0 container.__enter__ (test_run_command-dpp)
                - from drwxr-xr-x /home/runner/work/defects4cpp/defects4cpp/mydir/test_run_command0/test_run_command/yara/fixed#1
                - to /home/workspace
                - mode rw
INFO     processor.core.docker:(unknown file):0 container.__exit__ (test_run_command-dpp)
INFO     processor.core.docker:(unknown file):0 container.__enter__ (test_run_command-dpp)
                - from drwxr-xr-x /home/runner/work/defects4cpp/defects4cpp/mydir/test_run_command0/test_run_command/yara/fixed#1
                - to /home/workspace
                - mode rw
INFO     processor.core.docker:(unknown file):0 container.__exit__ (test_run_command-dpp)<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_test_command_callback.py::test_run_command_as_script</td>
          <td class="col-duration">21.91</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>  Starting container
    Image: hschoe/defects4cpp-ubuntu:test_run_command_as_script
Step 1/5 : FROM ubuntu:20.04

 ---&gt; 3bc6e9f30f51
Step 2/5 : RUN useradd --uid 1001 --home-dir /home/workspace --shell /bin/bash defects4cpp

 ---&gt; Using cache
 ---&gt; e33861a87ef0
Step 3/5 : USER defects4cpp

 ---&gt; Using cache
 ---&gt; aac010ec825b
Step 4/5 : ENV USER defects4cpp

 ---&gt; Using cache
 ---&gt; c1afafc3189e
Step 5/5 : WORKDIR /home/workspace

 ---&gt; Using cache
 ---&gt; 5794be79f75a
Successfully built 5794be79f75a
Successfully tagged hschoe/defects4cpp-ubuntu:test_run_command_as_script
  Closing container
  Starting container
    Image: hschoe/defects4cpp-ubuntu:test_run_command_as_script
  Closing container
<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     processor.core.docker:(unknown file):0 container.__enter__ (test_run_command_as_script-dpp)
                - from drwxr-xr-x /home/runner/work/defects4cpp/defects4cpp/mydir/test_run_command_as_script0/test_run_command_as_script/yara/fixed#1
                - to /home/workspace
                - mode rw
INFO     processor.core.docker:(unknown file):0 container.__exit__ (test_run_command_as_script-dpp)
INFO     processor.core.docker:(unknown file):0 container.__enter__ (test_run_command_as_script-dpp)
                - from drwxr-xr-x /home/runner/work/defects4cpp/defects4cpp/mydir/test_run_command_as_script0/test_run_command_as_script/yara/fixed#1
                - to /home/workspace
                - mode rw
INFO     processor.core.docker:(unknown file):0 container.__exit__ (test_run_command_as_script-dpp)<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_test_command_callback.py::test_additional_gcov_options</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_vcs.py::test_checkout_fixed</td>
          <td class="col-duration">2.56</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>[yara] cloning a new repository from &#x27;https://github.com/VirusTotal/yara&#x27;
[yara] checking out &#x27;cb787cc0c7edd91d245175af439a9190da97228b&#x27;
[yara] done
[yara] cloning a new repository from &#x27;https://github.com/VirusTotal/yara&#x27;
[yara] checking out &#x27;cb787cc0c7edd91d245175af439a9190da97228b&#x27;
[yara] done
<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     processor.checkout:(unknown file):0 git-clone &#x27;yara&#x27;
INFO     processor.checkout:(unknown file):0 cloning yara into /home/runner/work/defects4cpp/defects4cpp/mydir/test_checkout_fixed0/yara/.repo
INFO     processor.checkout:(unknown file):0 git-checkout
INFO     processor.checkout:(unknown file):0 git-am
INFO     processor.checkout:(unknown file):0 0001-split.patch
INFO     processor.checkout:(unknown file):0 creating &#x27;.defects4cpp.json&#x27; at /home/runner/work/defects4cpp/defects4cpp/mydir/test_checkout_fixed0/yara/fixed#1
INFO     processor.checkout:(unknown file):0 git-clone &#x27;yara&#x27;
INFO     processor.checkout:(unknown file):0 git-checkout
INFO     processor.checkout:(unknown file):0 git-am skipped
INFO     processor.checkout:(unknown file):0 creating &#x27;.defects4cpp.json&#x27; at /home/runner/work/defects4cpp/defects4cpp/mydir/test_checkout_fixed0/yara/fixed#1<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/processor/test_vcs.py::test_checkout_buggy</td>
          <td class="col-duration">2.51</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>[yara] cloning a new repository from &#x27;https://github.com/VirusTotal/yara&#x27;
[yara] checking out &#x27;cb787cc0c7edd91d245175af439a9190da97228b&#x27;
[yara] done
[yara] cloning a new repository from &#x27;https://github.com/VirusTotal/yara&#x27;
[yara] checking out &#x27;cb787cc0c7edd91d245175af439a9190da97228b&#x27;
[yara] done
<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     processor.checkout:(unknown file):0 git-clone &#x27;yara&#x27;
INFO     processor.checkout:(unknown file):0 cloning yara into /home/runner/work/defects4cpp/defects4cpp/mydir/test_checkout_buggy0/yara/.repo
INFO     processor.checkout:(unknown file):0 git-checkout
INFO     processor.checkout:(unknown file):0 git-am
INFO     processor.checkout:(unknown file):0 0001-split.patch, 0001-buggy.patch
INFO     processor.checkout:(unknown file):0 creating &#x27;.defects4cpp.json&#x27; at /home/runner/work/defects4cpp/defects4cpp/mydir/test_checkout_buggy0/yara/buggy#1
INFO     processor.checkout:(unknown file):0 git-clone &#x27;yara&#x27;
INFO     processor.checkout:(unknown file):0 git-checkout
INFO     processor.checkout:(unknown file):0 git-am skipped
INFO     processor.checkout:(unknown file):0 creating &#x27;.defects4cpp.json&#x27; at /home/runner/work/defects4cpp/defects4cpp/mydir/test_checkout_buggy0/yara/buggy#1<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/script/test_readme_generator.py::test_readme_generator</td>
          <td class="col-duration">0.01</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody></table></body></html>